spanner/row.go

```diff
@@ -378,3 +389,186 @@ func (r *Row) ToStructLenient(p interface{}) error {
 		true,
 	)
 }
+
+// SelectAll iterates all rows to the end. After iterating it closes the rows,
+// and propagates any errors that could pop up.
```

Comment:
This is a bit ambiguous: Is the destination slice half-filled if an error 'pops up'? Or is it then always empty? Could it happen that you get a filled destination slice and an error, and that the destination slice contains all data?

Reply:
Yes, error will be returned with destination partially filled, added a test case for the same.